### PR TITLE
fix: PoE replay support with HogQL

### PR DESCRIPTION
**Pull request by pauldambra**
we're still not using PoE and HogQL and person properties propertly maybe 🤷 

but this at least is closer - if still a little manual

### CODE REVIEW

File: ee/session_recordings/queries/test/test_session_recording_list_from_filters.py [hunk 1]
```diff
@@ -160,30 +187,58 @@ def test_effect_of_poe_settings_on_query_generated(
...
+                        "argMax(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(person.properties, %(hogql_val_6)s), ''), 'null'), '^\"|\"$', ''), person.version) AS properties___rgInternal"
+                        in printed_query
+                    )
+                else:
+                    # we should use materialized column
+                    # assert (
+                    #     "argMax(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(person.properties, %(hogql_val_6)s), ''), 'null'), '^\"|\"$', ''), person.version) AS properties___rgInternal"
+                    #     not in printed_query
+                    # )
+                    # TODO frustratingly this doesn't pass - but since we're migrating to PoE maybe we can ignore it
```
Comment (daibhin) [inline ee/session_recordings/queries/test/test_session_recording_list_from_filters.py:216]: Any idea why it doesn't pass? Do we know what timeline we're migrating to PoE on? I thought it was enabled for new users but didn't know if we planned to backfill

File: posthog/session_recordings/queries/session_recording_list_from_filters.py [hunk 2]
```diff
@@ -474,6 +484,9 @@ def _where_predicates(self) -> ast.Expr:
         if self.event_properties:
             exprs.append(property_to_expr(self.event_properties, team=self._team, scope="replay"))
 
+        if self._team.person_on_events_mode and self.person_properties:
```
Comment (daibhin) [inline posthog/session_recordings/queries/session_recording_list_from_filters.py:487]: Can we reuse the same `poe_mode_active` as above?

**Review by daibhin — APPROVED** [review]
Looks good. This file is really well tested so I'm also relatively confident if they pass

File: posthog/session_recordings/queries/session_recording_list_from_filters.py (hunk 2 above)
Comment (pauldambra) [reply posthog/session_recordings/queries/session_recording_list_from_filters.py:487]: yeah, we totally could 🙈 

i'll follow up once i know if it works, i'm too excited :)

**Pull request merged**